docs/presence_router_module.md

```diff
@@ -0,0 +1,222 @@
+# Presence Router Module
```

Comment:
I was asked today about how one would implement this module for an application service, instead of a user calling `/sync`, and had a bit of a look through the code. I noticed a couple of things:

1. There's some code that determines whether an application service is interested in a user or a room. This is used to check whether we need to consider sending an event linked to a room/user to the application service. We don't hook into this yet. I suppose we could add a call to `get_interested_users` here and call it with the `user_id` argument set to the Application Service's `sender_localpart` user? `get_users_for_states` would still be called to filter presence updates once an application service had been declared interested.

    This could also be worked around by having the AS expand their namespace regexes, thus making them interested in a wide range of users (potentially all), but I think that's a bit hacky.

2. The `ModuleApi.send_local_online_presence_to` method is currently only set up to modify `/sync` responses. I know we're trying to discourage AS users from syncing. So we probably need to modify the method to check if the user is a known application service `sender_localpart`, and if so send a burst of all current user presence states in a transaction.

I'll also add some tests for application services to verify what does and doesn't already work.

Reply:
After testing locally appservices do not currently receive presence updates.

However in the interest of not complicating this PR further, I'm going to leave a note about this and aim to fix it in a future PR.